pyannote/audio/tasks/segmentation/unsupervised_segmentation.py

```diff
@@ -325,3 +346,63 @@ def update_teacher_and_cache(
                 raise AttributeError(
                     f"TeacherUpdate callback can't be applied on this model : {err}"
                 )
+
+
+class BrouhahaPseudolabelsFilter(PseudoLabelPostprocess):
+    def __init__(
+        self,
+        model,
+        data: Literal["snr", "c50"],
+        mode: Literal["threshold", "quantile"],
+        threshold: float,
+        step_size_percent: float = 0.5,
+    ):
+        self.model = model
+        self.data = data
+        if self.data not in ["snr", "c50"]:
+            raise ValueError(f"Invalid data type ({data}), use snr or c50")
+        self.data_index = 1 if "snr" else 2
+        self.mode = mode
+        if self.mode not in ["threshold", "quantile"]:
+            raise ValueError(f"Invalid mode ({mode}), use threshold or quantile")
+        self.threshold = threshold
+        self.step_size_percent = step_size_percent
+
+    def compute_quantile(self, protocol, subset_name: str):
+        files = list(getattr(protocol, subset_name)())
+
+        (device,) = get_devices(needs=1)
+        inference = Inference(
+            self.model,
+            device=device,
+            duration=self.model.specifications.duration,
+            step=self.step_size_percent * self.model.specifications.duration,
+        )
+
+        means = torch.zeros(len(files))
+
+        for i in range(len(files)):
+            file = files[i]
```

Comment:
```suggestion
        for i, file in enumerate(files):
```

This is more Pythonic ;-)